### PR TITLE
Upgrade for TiFlash

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
Upgrade for TiFlash
### What is changed and how does it work?
Upgrade for TiFlash
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   - Update config for TiFlash and TiFlash statefulset will be rolling updated
   - If PD, TiKV, or TiDB are upgraded, TiFlash will wait for their rolling updates to be done first

Code changes

 - Has Go code change

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Support upgrading for TiFlash
```

### CODE REVIEW

File: pkg/manager/member/tiflash_upgrader.go [hunk 1]
```diff
@@ -38,9 +38,19 @@ func NewTiFlashUpgrader(pdControl pdapi.PDControlInterface,
...
+		tc.Status.TiDB.Phase == v1alpha1.UpgradePhase {
+		_, podSpec, err := GetLastAppliedConfig(oldSet)
+		if err != nil {
+			return err
+		}
+		newSet.Spec.Template.Spec = *podSpec
+		return nil
+	}
 
+	tc.Status.TiFlash.Phase = v1alpha1.UpgradePhase
```
Comment (weekface) [inline pkg/manager/member/tiflash_upgrader.go:53]: there are two places to set `tc.Status.TiFlash.Phase ` to `Upgrade`, can it be merged?

**Comment by DanielZhangQD** [comment]
@weekface comments addressed, PTAL.

**Pull request merged**